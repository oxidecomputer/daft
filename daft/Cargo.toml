[package]
name = "daft"
version = "0.1.0"
description = "Structural diffs of Rust data structures"
edition.workspace = true
repository.workspace = true
rust-version.workspace = true
license.workspace = true

[lints]
workspace = true

[dependencies]
daft-derive.workspace = true
newtype-uuid = { workspace = true, optional = true }
oxnet = { workspace = true, optional = true }
paste.workspace = true
uuid = { workspace = true, optional = true, features = ["v4"] }

[features]
newtype-uuid1 = ["dep:newtype-uuid"]
oxnet01 = ["dep:oxnet"]
uuid1 = ["dep:uuid"]
