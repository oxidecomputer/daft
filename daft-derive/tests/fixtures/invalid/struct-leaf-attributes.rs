use daft::Diffable;

#[derive(Diffable)]
#[daft(leaf)]
struct MyStruct {
    // Attributes on leaf-struct fields are banned.
    #[daft(ignore)]
    a: i32,
    #[daft(leaf)]
    b: i32,
}

fn main() {
    // MyStruct and its Diffable should exist.
    let before = MyStruct { a: 0, b: 0 };
    let after = MyStruct { a: 0, b: 1 };

    let _diff = before.diff(&after);
}
