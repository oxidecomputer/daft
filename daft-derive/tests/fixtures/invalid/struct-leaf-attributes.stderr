error: daft attributes are not allowed on fields of structs annotated with #[daft(leaf)]
 --> tests/fixtures/invalid/struct-leaf-attributes.rs:7:5
  |
7 |     #[daft(ignore)]
  |     ^^^^^^^^^^^^^^^

error: daft attributes are not allowed on fields of structs annotated with #[daft(leaf)]
 --> tests/fixtures/invalid/struct-leaf-attributes.rs:9:5
  |
9 |     #[daft(leaf)]
  |     ^^^^^^^^^^^^^
